common/src/constbuffer.c

```diff
@@ -0,0 +1,369 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <stdlib.h>
+#include <stdbool.h>
+#include "azure_macro_utils/macro_utils.h"
+#include "azure_c_logging/xlogging.h"
+#include "refcount.h"
+#include "gballoc.h"
+
+#include "constbuffer.h"
```

Comment:
I am maintaining the stance that constbuffer/gballoc thingies have nothing to do with PAL, but I guess that's case of disagreement and commit.